VstsTasks/AzureDtlCreateVM/task-funcs.ps1

```diff
@@ -205,18 +229,37 @@ function Validate-ArtifactStatus
     [CmdletBinding()]
     param(
         [string] $ResourceId,
+        [string] $TemplateName,
         [string] $Fail
     )
 
-    $fail = ConvertTo-Bool -Value $Fail
-    if ($fail)
+    $checkForFailedArtifacts = ConvertTo-Bool -Value $Fail
+    if ($checkForFailedArtifacts)
     {
-        $artifactDeploymentStatus = (Get-AzureRmResource -ResourceId $ResourceId -ODataQuery '$expand=Properties($expand=Artifacts)').Properties.artifactDeploymentStatus
-
-        [array]$failedArtifacts = $artifactDeploymentStatus | ? { $_.deploymentStatus -eq 'Failed' }
-        if ($failedArtifacts.Count -gt 0)
+        $templateFile = Get-TemplateFile -TemplateName $TemplateName
+        $armTemplateJson = [IO.File]::ReadAllText($templateFile) -replace '/\*([^\*/])*\*/',''

```

Comment:
Add a code comment as to what this is doing, which I assume is removing any code comments in the style of: /* ... */.

Also, perhaps we should not keep this in long term, eh?